### PR TITLE
CA-26/Firebase all the way

### PR DESCRIPTION
**Pull request by rock3r**
This is the first PR that reimplements the backend in Firebase. We start from the Schedule activity.

 * New viewmodels that are created from the API models directly (we don't need domain models)
 * Firebase in the repository implementation
    * Auto-refreshes data on changes, live (thanks to `DiffUtils`, even tho that needs to be moved off of the UI thread soon)
 * The progress bar is now hidden after data is retrieved
 * We use Novoda's `ViewPagerAdapter` for the schedule `ViewPager`
 * Bump `minSdkVersion` to 16 for greater justice (might rise further, likely be 21, soon)

firebaseeee

### CODE REVIEW

**Comment by rock3r** [comment]
h/t to @fourlastor, 99% of the code is his, but we finished it on my computer :D

File: app/src/main/java/com/connfa/schedule/service/ScheduleActivityService.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    public ScheduleActivityService(FirebaseConnfaRepository repository) {
+        this.repository = repository;
+    }
+
+    public Observable<Schedule> schedule() {
+        Observable<FirebaseEvent.Holder> eventObservable = repository.sessions();
+        Observable<FirebaseSpeaker.Holder> speakersObservable = repository.speakers();
+
+        return Observable.combineLatest(
+                eventObservable,
```
Comment (danybony) [inline app/src/main/java/com/connfa/schedule/service/ScheduleActivityService.java:31]: Why this bigger indentation?

**Review by tiwiz — APPROVED** [review]
LGTM

File: app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import io.reactivex.ObservableOnSubscribe;
+import io.reactivex.functions.Function;
+import io.reactivex.schedulers.Schedulers;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public final class FirebaseConnfaRepository {
+    private final DatabaseReference database;
+
+    public static FirebaseConnfaRepository newInstance(Context context) {
```
Comment (danybony) [inline app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java]: You don't need the context here

File: app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        DatabaseReference database = FirebaseDatabase.getInstance().getReference();
+        return new FirebaseConnfaRepository(database);
+    }
+
+    private FirebaseConnfaRepository(DatabaseReference database) {
+        this.database = database;
+    }
+
+    public Observable<FirebaseEvent.Holder> bofs() {
```
Comment (danybony) [inline app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java:36]: Brotherood of steel? 😆

**Review by danybony — CHANGES_REQUESTED** [review]
Great job, just a couple of comments

File: app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java (hunk 3 above)
Comment (tiwiz) [reply app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java:36]: I think it's exactly how it is called in the rest of the app, tho

Comment (danybony) [reply app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java:36]: Probably, but I am curious about the meaning

File: app/src/main/java/com/connfa/schedule/service/ScheduleActivityService.java (hunk 1 above)
Comment (rock3r) [reply app/src/main/java/com/connfa/schedule/service/ScheduleActivityService.java:31]: The continuation indent in the IDE settings is 8, it's correct

File: app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java (hunk 3 above)
Comment (tiwiz) [reply app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java:36]: Me too, I didn't find anything in the web. Maybe we should go to a Drupal Conf to know 😆

File: app/src/main/java/com/connfa/schedule/service/ScheduleActivityService.java (hunk 1 above)
Comment (danybony) [reply app/src/main/java/com/connfa/schedule/service/ScheduleActivityService.java:31]: 👎 for me, but not a blocker

File: app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java (hunk 3 above)
Comment (rock3r) [reply app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java:36]: Birds of Feather. Don't ask. I had no idea either.

File: app/src/main/java/com/connfa/schedule/service/ScheduleActivityService.java (hunk 1 above)
Comment (rock3r) [reply app/src/main/java/com/connfa/schedule/service/ScheduleActivityService.java:31]: It's the same we have in all projects man

**Comment by rock3r** [comment]
@danybony all comments addressed

**Review by danybony — APPROVED** [review]

**Pull request merged**